Automattic/wp-calypso
# Date Control: Extract stats date control into a reusable component to keep consistency.

Related to https://github.com/Automattic/dotcom-forge/issues/9770

## Proposed Changes
As part of the goal to make design across calypso more consistent we are extracting the [global date picker ](puPv3-gtX-p2) used in Jetpack stats into a reusable `DateControl` component that will be able to used in the sites log page while maintaining the styling and behavior.

* Extract the styling and reusable parts of `StatsDateControl` into `DateControl`
* Modify `StatsDateControl` to use the new `DateControl` component while keeping the parts unique to the `StatsDateControl`

![image](https://github.com/user-attachments/assets/f05a0eaa-9c5c-4584-b199-a78e691883bf)


## Why are these changes being made?


* As part of keeping consistency across calypso 

## Testing Instructions



* Go to `/stats/day/siteSlug` and ensure the date control works as expected and nothing breaks

## Pre-merge Checklist
